queue.c

```diff
@@ -60,11 +68,19 @@ queue_init(queue_t *queue) {
 
 bool
 queue_empty(queue_t *queue) {
+    if (queue == NULL) {
+        return true;
```

Comment:
I'm not sure if `true` is the sensible default here. It makes more sense than `false`, as a non existent queue is more empty than full, but it does not inform the caller that an error occured.

Reply:
Since the other functions (returning `void`) do not report an error to the user either when recieving a `NULL` value this should be fine for now.

Also this function is only called internally (and will be made properly private soon), after another function checked for a `NULL` value already, so currently this will never be called with `NULL`.

Resolving this now.